lib/module.js

```diff
@@ -144,52 +182,24 @@ Module._findPath = function(request, paths, isMain) {
     return Module._pathCache[cacheKey];
   }
 
-  var exts;
+  const exts = Object.keys(Module._extensions);
   const trailingSlash = request.length > 0 &&
                         request.charCodeAt(request.length - 1) === 47/*/*/;
+  const isAbsolute = !trailingSlash;
 
   // For each path
   for (var i = 0; i < paths.length; i++) {
     // Don't search further if path doesn't exist
-    const curPath = paths[i];
+    var curPath = paths[i];
```

Comment:
I test again on const in loop, it will be `TurboFan compiler`

```bash
node v6.0.0
function testFn(a, b) {
  const foo = [1, 2, 3];
  let total = 0;
  for (const a of foo) {
    const c = a;
    total += c;
  }
  return total;
}
[disabled optimization for 0x3a6acdbf0661 <SharedFunctionInfo formatValue>, reason: TryCatchStatement]
6
[didn't find optimized code in optimized code map for 0x4dca4458ee9 <SharedFunctionInfo testFn>]
[compiling method 0x3b482c0c0669 <JS Function testFn (SharedFunctionInfo 0x4dca4458ee9)> using TurboFan]
[optimizing 0x3b482c0c0669 <JS Function testFn (SharedFunctionInfo 0x4dca4458ee9)> - took 1.139, 0.000, 0.000 ms]
Function is TurboFan compiler
```